### DISCUSSION TITLE
Stack arrays in modules

### DISCUSSION THREAD

**Discussion by EmilyBourne** [pyccel/pyccel#1146] [Ideas]
I would like to be able to have a stack array in a module. Currently stack arrays are created by decorators on the function therefore it is not possible to have a stack array in a module

**Comment by yguclu** [comment]
Talking about how to expose this to the user, I see two different ways:
  1. Using a comment with a specific pragma syntax
  2. Calling a special function from the Pyccel library

Option 1 seems less invasive to me, but potentially trickier to implement if we allow such a comment to appear anywhere in the code.

**Reply by EmilyBourne** [reply]
This could also be managed with type hinting ( #1462 )

**Reply by yguclu** [reply]
Yes! Good idea!
